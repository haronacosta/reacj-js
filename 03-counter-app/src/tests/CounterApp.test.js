import React from 'react'
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas basicas en CounterApp', () => {
   
    let wrapper =  shallow(<CounterApp />);


    // beforeEach se ejecuta antes de cada prueba 
    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    })


    test('Mostrar <CounterApp /> y sus valores por defecto', () => {
        
        const valorDefecto = wrapper.find('h2').text();

        expect(wrapper).toMatchSnapshot();

        expect(valorDefecto).toBe("10")
    });

    test('Mostrar <CounterApp /> y su valor por defecto debe de ser 100', () => {
        const valor = 100;

        const wrapper = shallow(<CounterApp value={valor} />);

        const valorDefecto = parseInt( wrapper.find('h2').text());

        expect(valorDefecto).toBe(valor)
    });

    test('Pruebas al boton de +1', () => {

        wrapper.find('button').at(0).simulate('click');


         const valor = wrapper.find('h2').text();

         expect(valor).toBe('11');
    });

    test('Pruebas al boton de -1', () => {

        wrapper.find('button').at(2).simulate('click');
    
        const valor = wrapper.find('h2').text();

         expect(valor).toBe('9');
    });
    
    test('Pruebas al boton de reset', () => {
        const valor = 105;

        const wrapper = shallow(<CounterApp value={valor} />);

        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');

        wrapper.find('button').at(1).simulate('click');
    
        const valorObtenido = wrapper.find('h2').text();

         expect(valorObtenido).toBe(`${valor}`);
    });
    
    
})
