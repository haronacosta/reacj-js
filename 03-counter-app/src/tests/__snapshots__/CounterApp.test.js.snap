// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas basicas en CounterApp Mostrar <CounterApp /> y su valor por defecto debe de ser 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`Pruebas basicas en CounterApp Mostrar <CounterApp /> y sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
